src/main/java/com/tabnine/binary/BinaryProcessGateway.java

```diff
@@ -36,4 +36,8 @@ public void writeRequest(String request) throws IOException {
     public boolean isDead() {
         return process == null || !process.isAlive();
     }
+
+    public void destroy() {
+        process.destroy();
```

Comment:
process can't be null?
What happens if destroy throws an exception (don't know if that can happen)?

Reply:
good catch

Reply:
Nope it can't. `init` is called

Reply:
We check for null in isDead so is the check redundant there or missing here?